### ISSUE TITLE
False positive with --warn-unreachable: TypedDict may be empty

### ISSUE THREAD

**Issue by ilai-deutel** [python/mypy#7743]
Hi,

I believe there is a false positive related to empty `TypedDict`s with `total=False`.

Using `--warn-unreachable`:

```python
from typing_extensions import TypedDict

class A(TypedDict, total=False):
    ...

class B(TypedDict, total=False):
    x: int

a: A = {}
b: B = {}

if not a:
    print('a is empty')

if not b:
    print('b is empty')  # error: Statement is unreachable

```

Using Python 3.7.4, tested with both mypy 0.740 and mypy master.

**Comment by ilevkivskyi** [comment]
Thanks for reporting! This looks like something easy to fix. In the `TypedDictType` constructor in `mypy/types.py` one just needs to use `required_keys` instead of `items` to determine value of `self.can_be_false`.
